[FEATURE] Migrating to lingui

crysfel opened the pull request:
# Details

## Description

Still not ready, I need to copy translations for Spanish and French. But is currently displaying all text from the i18n files using lingui! The CLI is great for extracting keys :)

crysfel commented on src/components/footer/DonateButton.js:
```diff
@@ -19,11 +17,11 @@ function DonateButton({ t }) {
         border="0"
         name="submit"
-        alt={t('footer-donate-paypal')}
+        alt={'footer-donate-paypal'}
```
We are not using this component yet, therefore I didn't set the i18n yet. When putting it back, we should set this value.

crysfel commented on src/components/footer/__tests__/__snapshots__/Footer.test.js.snap:
```diff
@@ -13,17 +13,17 @@ exports[`renders the footer element 1`] = `
       class="button tertiary small rounded capitalize"
     >
-      Terms & Conditions
+      footer.legal.terms
```
Instead of testing for the actual test, we are now testing the keys. Reason for this is in our test the `I18nProvider` is not defined, I feel like we should not add it. Also.... I think we should remove snapshot testing... but that's another topic xD

zsid commented on src/config/i18n/en/messages.js:
```diff
@@ -0,0 +1,49 @@
+  },
+  messages: {
+    'common.close': 'Close',
+    'footer.legal.cookies': 'Cookies',
```
what if we have it as an object for all? It will be easier to insert things there i.e.
```
messages: {
  footer: {
    legal: {  cookies: 'Cookies', privacy: 'Privacy and Policy  } etc
  }
}
```

crysfel replied:
This file is generated, maybe we shouldn't store it in git. But I will have to add another task to the `start` task to compile the translations before starting webpack.

I think having it here is confusing, I will go ahead an remove it.

crysfel replied:
Lingui doesn't allows to customize the keys in this way. We can use other format such as `po`: https://lingui.js.org/ref/conf.html#std:config-format

zsid replied:
Sounds good. As long as the file is not generated by us, it is fine 😄

zsid commented on src/config/i18n/en/messages.json:
```diff
@@ -0,0 +1,20 @@
+{
```
same the comment in `src/config/i18n/en/messages.js` about having them as objects

crysfel replied:
same as my previous reply 😂

zsid commented on src/config/i18n/es/messages.js:
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable */ module.exports = {
```
same as `src/config/i18n/en/messages.js`

crysfel replied:
same as my previous reply 😂

zsid commented on src/pages/home/Home.js:
```diff
@@ -41,59 +43,57 @@ class Home extends Component {
+                      target="_blank"
+                      rel="noopener noreferrer"
+                    >
+                      Slack Organization
```
do we want a variable fort this too?

crysfel replied:
Yeah, it gets replaced by the macro at build time (Line 73)

crysfel commented on src/components/button/Button.js:
```diff
@@ -38,7 +38,7 @@ Button.propTypes = {
   typography: PropTypes.oneOf(['uppercase', 'capitalize', 'lowercase', 'none']),
   shape: PropTypes.oneOf(['square', 'rounded', 'pill']),
-  children: PropTypes.node.isRequired,
+  children: PropTypes.node,
```
Agreed, but there's a warning in the console because content is null before the translation are loaded :( that's why I removed the `required`. Are we ok with that warning there?

zsid replied:
hmmm. Should we display anything on the page of there is no `content`?

crysfel commented:
@zsid thanks for taking the time to review. I'm going to update a couple things to avoid getting confused by the dictionaries.

But, the source are actually the JSON files (src/config/i18n/messages.json), the JS files (src/config/i18n/messages.js)  are auto generated by the lingui CLI.

zsid commented:
Is it worth updating the README about `lingui`?

crysfel commented:
@zsid absolutely! I've added some documentation to the readme file, removed the compiled files and introduced `npm-run-all` to run several task in sequence.

Please take a look!
